### PR TITLE
ONNX CastLike operator tests

### PR DESCRIPTION
**Pull request by tomdol**
### Details:
 - completion of https://github.com/openvinotoolkit/openvino/pull/13045

### Tickets:
 - 82470

### CODE REVIEW

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 1]
```diff
@@ -6141,3 +6141,133 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_constant_of_shape_null_node) {
...
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, castlike_float64_to_int64) {
+    auto function = onnx_import::import_onnx_model(file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                                        SERIALIZED_ZOO,
+                                                                        "onnx/castlike_float64_to_int64.onnx"));
+
+    auto test_case = test::TestCase(function, s_device);
+
+    test_case.add_input<float>(Shape{1, 1, 2, 2}, std::vector<float>{1.5, 2.3, 3, 4});
```
Comment (mbencer) [inline src/frontends/onnx/tests/onnx_import.in.cpp:6194]: shouldn't be `double` type here?

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 2]
```diff
@@ -6141,3 +6141,133 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_constant_of_shape_null_node) {
...
+    auto function = onnx_import::import_onnx_model(file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                                        SERIALIZED_ZOO,
+                                                                        "onnx/castlike_bfloat16_to_float32.onnx"));
+
+    auto test_case = test::TestCase(function, s_device);
+
+    test_case.add_input<bfloat16>(
+        Shape{3, 4},
+        std::vector<bfloat16>{121.5, 122.7, 3, 4, 5, 6, 7, 8.8, 9, 10, 11, 12});  // 122.7 is not 122.5, might be a bug,
+                                                                                  // create a ticket in JIRA for CPU
```
Comment (mbencer) [inline src/frontends/onnx/tests/onnx_import.in.cpp:6267]: do we have a ticket for it?

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 3]
```diff
@@ -6141,3 +6141,133 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_constant_of_shape_null_node) {
...
+                                                                                  // team
+    test_case.add_input<float>(Shape{3, 4}, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12});
+    test_case.add_expected_output<float>(std::vector<float>{121.5, 122.7, 3, 4, 5, 6, 7, 8.75, 9, 10, 11, 12});
+
+    test_case.run();
+}
```
Comment (mbencer) [inline src/frontends/onnx/tests/onnx_import.in.cpp:6273]: nit: EOF

**Review by mbencer — APPROVED** [review]
LGTM with some minor comments

File: src/frontends/onnx/tests/onnx_import.in.cpp (hunk 2 above)
Comment (tomdol) [reply src/frontends/onnx/tests/onnx_import.in.cpp]: I have to verify and create one if it doesn't exist yet. I've removed the comment though.

**Pull request merged**